nn/math_.py

```diff
@@ -27,6 +27,11 @@ def gelu(x: LayerRef) -> Layer:
   return activation(x, activation="gelu")
 
 
+def glu(x: LayerRef) -> Layer:
+  """GLU https://arxiv.org/abs/1612.08083"""
+  return activation(x, activation='glu')
```

Comment:
I just realize what this is. We don't have that on RETURNN side, and also, it would not be via the `ActivationLayer` because this would change the output dimension.

You could implement this here by hand using `nn.split`.

Reply:
hmm yes it is wrong to use activation class. However, we have gating in RETURNN. It is the `GatingLayer`. But I just noticed that it is in the blacklist here. PyTorch supports it as [here](https://pytorch.org/cppdocs/api/function_namespacetorch_1_1nn_1_1functional_1a789cc6940523640f299df85a3dc5a0fd.html). [Here](https://github.com/espnet/espnet/blob/7a1329a6fb713a17f5860ea7aed2315ade8d9860/espnet/nets/pytorch_backend/conformer/convolution.py#L71) an example how it is used in ESPNet.

Reply:
I blacklisted it because it is a bit too less generic (on RETURNN side) and it would be very trivial to reimplement it anyway in a generic way here on returnn-common side. Sth like:
```
def glu(x):
  a, b = nn.split(x, 2)
  return a * nn.sigmoid(b)
```